src/components/FeatureViews/Accounts/TabComponents/InstructorSchedule.js

```diff
@@ -1,130 +1,124 @@
 
 const InstructorSchedule = ({instructorID}) => {
-	const sessions = useSelector(({Calendar}) => Calendar.CourseSessions);
-	const instructor = useSelector(
-		({Users}) => Users.InstructorList[instructorID]
-	);
-	const OOOstatus = hooks.useOutOfOffice();
-	const availabilityStatus = hooks.useInstructorAvailability(instructorID);
-	const classEnrollmentStatus = hooks.useClassSessionsInPeriod("week");
-	const tutoringEnrollmentStatus = hooks.useTutoringSessionsInPeriod("week");
-
-	const teachingSessions = useMemo(
-		() =>
-			Object.values(sessions[instructorID] || {}).map((session) => ({
-				end: new Date(session.end_datetime),
-				start: new Date(session.start_datetime),
-				title: session.title,
-			})),
-		[sessions, instructorID]
-	);
 
-	const OOO = useMemo(
-		() =>
-			instructor
-				? Object.values(instructor.schedule.time_off).map(
-				({all_day, description, start, end}) => {
-					const endDate = new Date(end);
-					// since the end date for allDay events is EXCLUSIVE
-					// must add one day to include the end specified by user
-					if (all_day) {
-						endDate.setDate(endDate.getDate() + 1);
-					}
-					return {
-						allDay: all_day,
-						end: endDate,
-						start,
-						title: description || "Out of Office",
-					};
-				}
-				)
-				: [],
-		[instructor]
-	);
 
-	const allDayOOO = useMemo(() => OOO.some(({allDay}) => allDay), [OOO]);
-
-	const hoursWorked = useMemo(
-		() =>
-			Math.round(
-				Object.values(sessions[instructorID] || []).reduce((hours, session) => {
-					const start = new Date(session.start_datetime);
-					const end = new Date(session.end_datetime);
-					if (start > Date.now()) {
-						return hours;
-					} else if (end < Date.now()) {
-						return hours + toHours(end - start);
-					}
-					return hours + toHours(end - Date.now());
-				}, 0) * 100
-			) / 100,
-		[sessions, instructorID]
-	);
+	const { loading, error, data } = useQuery(GET_INSTRUCTOR_INFO, {
+		variables: {instructorID}
+	})
 
-	const instructorBusinessHours = useMemo(
-		() =>
-			instructor
-				? Object.values(instructor.schedule.work_hours).map(
-				({day, start, end}) => ({
-					daysOfWeek: [day],
-					endTime: end,
-					startTime: start,
-				})
-				)
-				: [],
-		[instructor]
-	);
+	if (loading ) return null;
+
+	if (error ) return "Unable to load schedule";
 
-	if (
-		teachingSessions.length === 0 &&
-		instructorBusinessHours.length === 0 &&
-		OOO.length === 0
-	) {
-		if (
-			hooks.isLoading(
-				tutoringEnrollmentStatus,
-				OOOstatus,
-				classEnrollmentStatus,
-				availabilityStatus
-			)
-		) {
-			return <Loading/>;
-		} else if (
-			hooks.isFail(
-				tutoringEnrollmentStatus,
-				OOOstatus,
-				classEnrollmentStatus,
-				availabilityStatus
-			)
-		) {
-			return "Unable to load schedule!";
+	const { user } = data.instructor;
+
+	const today = new Date(Date.now());
+
+	const hoursWorkedThisMonth = parseInt(user.instructor.sessionSet.reduce((hours, session) => {
```

Comment:
does this work? like under the user object there's an instructor key? if this is the case then you may want to destructure line 58 to be like: `const { user: { instructor } } = data. instructor` or `const { instructor } = data.instructor.user`

Reply:
I need access to the user object, so I changed it to `const { user, user : { instructor } } = data.instructor; `